frontend/src/components/vuetify/NavigationDrawer.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <v-navigation-drawer
+    v-model="drawer"
+    :location="computedLocation"
+    color="grey-lighten-4"
+    temporary
+    width="400"
+    class="custom-drawer"
+    mobile
+    mobile-breakpoint="md"
+  >
+    <slot />
+  </v-navigation-drawer>
+</template>
+
+<script lang="ts" setup>
+import { computed, ref, watch, PropType } from 'vue'
+import { useDisplay } from 'vuetify'
+
+type ValidLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'
+
+const props = defineProps({
+  modelValue: {
+    type: Boolean,
+    required: true,
+  },
+  location: {
+    type: String as PropType<NavigationDrawerLocation>,
+    required: false,
+    default: 'right',
+    validator: (value: string) =>
+      ['right', 'bottom', 'left', 'end', 'top', 'start'].includes(value as ValidLocation),
+  },
+})
```

Comment:
I think my suggestion here from last time went wrong. I meant:

```suggestion
export type DrawerLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: DrawerLocation
  }>(),
  {
    location: 'right',
  },
)
```

I am fully aware that Vuetify internally does this in a different way (more like you do), but I think that's because they provide a library which could also be used by pure JS apps without Typescript support, and in that case it's the safer way to do it. And probably they still haven't updated their code to the new prop types notation.

Reply:
code adopted